crates/uv-distribution/src/distribution_database.rs

```diff
@@ -449,7 +487,17 @@ impl<'a, Context: BuildContext> DistributionDatabase<'a, Context> {
                 // Download and unzip the wheel to a temporary directory.
                 let temp_dir = tempfile::tempdir_in(self.build_context.cache().root())
                     .map_err(Error::CacheWrite)?;
-                uv_extract::stream::unzip(&mut hasher, temp_dir.path()).await?;
+
+                match self.reporter {
+                    Some(ref reporter) => {
+                        let mut reader =
+                            ProgressReader::new(&mut hasher, progress.unwrap(), &**reporter);
```

Comment:
I feel like I'm so allergic to unwrap that I'd probably do something like:

```rust
let progress = self
    .reporter
    .as_ref()
    .map(|reporter| (reporter, reporter.on_download_start(dist.name(), size)));
```

And then use `match self.progress` everywhere. What's your instinct?